### ISSUE TITLE
报错：generate() argument after ** must be a mapping, not Tensor，transformers包已经更新到4.41.2

### ISSUE THREAD

**Issue by NEALKARS** [THUDM/GLM-4#410]
### System Info / 系統信息

Package                      Version
---------------------------- ---------------
absl-py                      2.1.0
accelerate                   0.33.0
aiofiles                     23.2.1
aiohttp                      3.9.5
aiosignal                    1.3.1
altair                       5.3.0
annotated-types              0.6.0
anyio                        4.3.0
argon2-cffi                  23.1.0
argon2-cffi-bindings         21.2.0
arrow                        1.3.0
arxiv                        2.1.0
asttokens                    2.4.1
astunparse                   1.6.3
async-lru                    2.0.4
async-timeout                4.0.3
attrs                        23.2.0
Babel                        2.15.0
beautifulsoup4               4.12.3
bitsandbytes                 0.43.2
bleach                       6.1.0
blinker                      1.8.2
blobfile                     2.1.1
Brotli                       1.0.9
cachetools                   5.3.3
certifi                      2024.2.2
cffi                         1.16.0
charset-normalizer           3.3.2
click                        8.1.7
comm                         0.2.2
contourpy                    1.2.1
cpm-kernels                  1.0.11
cycler                       0.12.1
dataclasses-json             0.6.5
datasets                     2.19.1
debugpy                      1.8.1
decorator                    5.1.1
deepspeed                    0.13.1
defusedxml                   0.7.1
dill                         0.3.8
distro                       1.9.0
dnspython                    2.6.1
einops                       0.8.0
email_validator              2.1.1
exceptiongroup               1.2.1
executing                    2.0.1
fairscale                    0.4.13
fastapi                      0.111.0
fastapi-cli                  0.0.2
fastjsonschema               2.19.1
feedparser                   6.0.10
ffmpy                        0.3.2
filelock                     3.14.0
fire                         0.6.0
flatbuffers                  24.3.25
fonttools                    4.51.0
fqdn                         1.5.1
frozenlist                   1.4.1
fsspec                       2024.3.1
gast                         0.5.4
gitdb                        4.0.11
GitPython                    3.1.43
google-pasta                 0.2.0
gradio                       4.39.0
gradio_client                1.1.1
greenlet                     3.0.3
grpcio                       1.63.0
h11                          0.14.0
h5py                         3.11.0
hjson                        3.1.0
httpcore                     1.0.5
httptools                    0.6.1
httpx                        0.27.0
huggingface-hub              0.24.2
idna                         3.7
importlib_resources          6.4.0
ipykernel                    6.29.4
ipython                      8.24.0
ipywidgets                   8.1.2
isoduration                  20.11.0
jedi                         0.19.1
jieba                        0.42.1
Jinja2                       3.1.3
joblib                       1.4.2
json5                        0.9.25
jsonpatch                    1.33
jsonpointer                  2.4
jsonschema                   4.22.0
jsonschema-specifications    2023.12.1
jupyter                      1.0.0
jupyter_client               8.6.1
jupyter-console              6.6.3
jupyter_core                 5.7.2
jupyter-events               0.10.0
jupyter-lsp                  2.2.5
jupyter_server               2.14.0
jupyter_server_terminals     0.5.3
jupyterlab                   4.1.8
jupyterlab_pygments          0.3.0
jupyterlab_server            2.27.1
jupyterlab_widgets           3.0.10
keras                        3.3.3
kiwisolver                   1.4.5
langchain                    0.1.17
langchain-community          0.0.37
langchain-core               0.1.52
langchain-text-splitters     0.0.1
langchainhub                 0.1.15
langsmith                    0.1.54
latex2mathml                 3.77.0
libclang                     18.1.1
loguru                       0.7.2
lxml                         4.9.4
Markdown                     3.6
markdown-it-py               3.0.0
MarkupSafe                   2.1.5
marshmallow                  3.21.2
matplotlib                   3.8.4
matplotlib-inline            0.1.7
mdtex2html                   1.3.0
mdurl                        0.1.2
mistune                      3.0.2
ml-dtypes                    0.3.2
mpi4py                       3.1.6
mpmath                       1.3.0
multidict                    6.0.5
multiprocess                 0.70.16
mypy-extensions              1.0.0
namex                        0.0.8
nbclient                     0.10.0
nbconvert                    7.16.4
nbformat                     5.10.4
nest-asyncio                 1.6.0
networkx                     3.3
ninja                        1.11.1.1
nltk                         3.8.1
notebook                     7.1.3
notebook_shim                0.2.4
numpy                        1.26.4
nvidia-cublas-cu12           12.1.3.1
nvidia-cuda-cupti-cu12       12.1.105
nvidia-cuda-nvrtc-cu12       12.1.105
nvidia-cuda-runtime-cu12     12.1.105
nvidia-cudnn-cu12            8.9.2.26
nvidia-cufft-cu12            11.0.2.54
nvidia-curand-cu12           10.3.2.106
nvidia-cusolver-cu12         11.4.5.107
nvidia-cusparse-cu12         12.1.0.106
nvidia-nccl-cu12             2.20.5
nvidia-nvjitlink-cu12        12.4.127
nvidia-nvtx-cu12             12.1.105
openai                       1.37.1
opt-einsum                   3.3.0
optree                       0.11.0
orjson                       3.10.3
overrides                    7.7.0
packaging                    23.2
pandas                       2.2.2
pandocfilters                1.5.1
parso                        0.8.4
peft                         0.10.0
pexpect                      4.9.0
pillow                       10.3.0
pip                          23.3.1
platformdirs                 4.2.1
prometheus_client            0.20.0
prompt-toolkit               3.0.43
protobuf                     4.25.3
psutil                       5.9.8
ptyprocess                   0.7.0
pure-eval                    0.2.2
py-cpuinfo                   9.0.0
pyarrow                      16.0.0
pyarrow-hotfix               0.6
pycparser                    2.22
pycryptodomex                3.20.0
pydantic                     2.7.1
pydantic_core                2.18.2
pydeck                       0.9.0
pydub                        0.25.1
Pygments                     2.18.0
PyJWT                        2.8.0
pynvml                       11.5.0
pyparsing                    3.1.2
PySocks                      1.7.1
python-dateutil              2.9.0.post0
python-dotenv                1.0.1
python-json-logger           2.0.7
python-multipart             0.0.9
pytz                         2024.1
PyYAML                       6.0.1
pyzmq                        26.0.3
qtconsole                    5.5.2
QtPy                         2.4.1
referencing                  0.35.1
regex                        2024.4.28
requests                     2.31.0
rfc3339-validator            0.1.4
rfc3986-validator            0.1.1
rich                         13.7.1
rouge-chinese                1.0.3
rpds-py                      0.18.1
ruamel.yaml                  0.18.6
ruamel.yaml.clib             0.2.8
ruff                         0.4.3
safetensors                  0.4.3
scikit-learn                 1.4.2
scipy                        1.13.0
semantic-version             2.10.0
Send2Trash                   1.8.3
sentence-transformers        2.7.0
sentencepiece                0.2.0
setuptools                   69.5.1
sgmllib3k                    1.0.0
shellingham                  1.5.4
six                          1.16.0
smmap                        5.0.1
sniffio                      1.3.1
soupsieve                    2.5
SQLAlchemy                   2.0.30
sse-starlette                2.1.0
stack-data                   0.6.3
starlette                    0.37.2
streamlit                    1.34.0
sympy                        1.12
tenacity                     8.3.0
tensorboard                  2.16.2
tensorboard-data-server      0.7.2
tensorflow                   2.16.1
tensorflow-io-gcs-filesystem 0.37.0
termcolor                    2.4.0
terminado                    0.18.1
threadpoolctl                3.5.0
tiktoken                     0.7.0
timm                         0.9.16
tinycss2                     1.3.0
tokenizers                   0.19.1
toml                         0.10.2
tomli                        2.0.1
tomlkit                      0.12.0
toolz                        0.12.1
torch                        2.3.0
torchvision                  0.18.0
tornado                      6.4
tqdm                         4.66.2
traitlets                    5.14.3
transformers                 4.41.2
triton                       2.3.0
typer                        0.12.3
types-python-dateutil        2.9.0.20240316
types-requests               2.31.0.20240406
typing_extensions            4.11.0
typing-inspect               0.9.0
tzdata                       2024.1
ujson                        5.9.0
uri-template                 1.3.0
urllib3                      2.2.1
uvicorn                      0.29.0
uvloop                       0.19.0
watchdog                     4.0.0
watchfiles                   0.21.0
wcwidth                      0.2.13
webcolors                    1.13
webencodings                 0.5.1
websocket-client             1.8.0
websockets                   11.0.3
Werkzeug                     3.0.2
wheel                        0.43.0
widgetsnbextension           4.0.10
wrapt                        1.16.0
xxhash                       3.4.1
yarl                         1.9.4
zhipuai                      2.0.1.2024042

### Information / 问题信息

- [ ] The official example scripts / 官方的示例脚本
- [ ] My own modified scripts / 我自己修改的脚本和任务

### Reproduction / 复现过程

直接使用官方网址提供的代码段跑的，模型可以加载，但是到最后model.generate那边就报错：
import torch
from transformers import AutoModelForCausalLM, AutoTokenizer

device = "cuda"

tokenizer = AutoTokenizer.from_pretrained("THUDM/glm-4-9b-chat", trust_remote_code=True)

query = "你好"

inputs = tokenizer.apply_chat_template([{"role": "user", "content": query}],
                                       add_generation_prompt=True,
                                       tokenize=True,
                                       return_tensors="pt",
                                       return_dict=True
                                       )

inputs = inputs.to(device)
model = AutoModelForCausalLM.from_pretrained(
    "THUDM/glm-4-9b-chat",
    torch_dtype=torch.bfloat16,
    low_cpu_mem_usage=True,
    trust_remote_code=True
).to(device).eval()

gen_kwargs = {"max_length": 2500, "do_sample": True, "top_k": 1}
with torch.no_grad():
    outputs = model.generate(**inputs, **gen_kwargs)
    outputs = outputs[:, inputs['input_ids'].shape[1]:]
    print(tokenizer.decode(outputs[0], skip_special_tokens=True))

### Expected behavior / 期待表现

通过

**Comment by NEALKARS** [comment]
已解决，仓库代码更新到最新，requirement重新装一下。

**Comment by echoht** [comment]
> 已解决，仓库代码更新到最新，requirement重新装一下。

你好，是哪个仓库的代码更新和requirement重装呢？
